homeassistant/components/fyta/sensor.py

```diff
@@ -0,0 +1,183 @@
+        device_class=SensorDeviceClass.ENUM,
+        options=PLANT_STATUS_LIST,
+        icon="mdi:thermometer-lines",
+    ),
+    FytaSensorEntityDescription(
+        key="light_status",
+        translation_key="fyta_light_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=PLANT_STATUS_LIST,
+        icon="mdi:sun-clock-outline",
+    ),
+    FytaSensorEntityDescription(
+        key="moisture_status",
+        translation_key="fyta_moisture_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=PLANT_STATUS_LIST,
+        icon="mdi:water-percent-alert",
+    ),
+    FytaSensorEntityDescription(
+        key="salinity_status",
+        translation_key="fyta_salinity_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=PLANT_STATUS_LIST,
+        icon="mdi:sprout-outline",
+    ),
+    FytaSensorEntityDescription(
+        key="temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="light",
+        translation_key="fyta_light",
+        native_unit_of_measurement="mol/d",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:weather-sunny",
+    ),
+    FytaSensorEntityDescription(
+        key="moisture",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.MOISTURE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="salinity",
+        translation_key="fyta_salinity",
+        native_unit_of_measurement="mS/cm",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:sprout-outline",
+    ),
+    FytaSensorEntityDescription(
+        key="ph",
+        device_class=SensorDeviceClass.PH,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    FytaSensorEntityDescription(
+        key="battery_level",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    FytaSensorEntityDescription(
+        key="plant_number",
+        translation_key="fyta_plant_number",
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        icon="mdi:update",
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the FYTA binary sensors."""
+    coordinator: FytaCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    plant_entities: list[dict[str, Any]] = []
+    plant_entities.extend(
+        [
+            FytaCoordinatorSensor(coordinator, entry, sensor)
+            for sensor in SENSORS
+            if sensor.key in coordinator.data
+        ]
+    )
```

Comment:
```suggestion
    plant_entities: list[dict[str, Any]] = [
            FytaCoordinatorSensor(coordinator, entry, sensor)
            for sensor in SENSORS
            if sensor.key in coordinator.data
        ]
```

The type doesn't seem right to me as well

Reply:
You are correct. I accepted the adjustment proposal of @joostlek without noticing, that it combined the list for FytaCoordinatorSensor and FytaEntitySensor, which both have different constructors. I suggest the following code (note, that based on an input above, FytaEntitySensor now also inherits from CoordinatorEntity):
```
    plant_entities: list[CoordinatorEntity] = []
    plant_entities.extend(
        [
            FytaCoordinatorSensor(coordinator, entry, sensor)
            for sensor in SENSORS
            if sensor.key in coordinator.data
        ]
    )

    plants = coordinator.plant_list.keys()
    plant_entities.extend(
        [
            FytaPlantSensor(coordinator, entry, sensor, plant_id)
            for plant_id in plants
            for sensor in SENSORS
            if sensor.key in coordinator.data[plant_id]
        ]
    )
```

Reply:
Looks good!

Reply:
or well, you could replace `plants` in the second iteration with just `coordinator.plant_list`

Reply:
From my perspective the first extend is unnecessary, you can assign the variable directly to the first list comprehension and then extend the second list comprehension.

Reply:
> or well, you could replace `plants` in the second iteration with just `coordinator.plant_list`

Would this really work? I need the key (where the plant_id is stored) and not only the value. How could I get the key of plant_id (that obviously would have to be renamed) then?

Reply:
```py
for k in adict.keys():
```
is the same as
```py
for k in adict:
```

Reply:
Thank you for the explenation! I was of the impression that the pair and not the key was assigned. I will change accordingly.